Core/clim-basic/extended-streams/recording.lisp

```diff
@@ -142,45 +142,41 @@ all of FUNCTION-ARGS as APPLY arguments."
 ;;; methods are very similar, hence this macro.  In order to exploit
 ;;; the similarities, it's necessary to treat the slots of the second
 ;;; record like variables, so for convenience the macro will use
-;;; SLOT-VALUE on both records.
-
+;;; WITH-SLOTS on the second record.
 (defmacro defrecord-predicate (record-type slots &body body)
-  "Each element of SLOTS is either a symbol or (:initarg-name slot-name)."
-  (let* ((slot-names (mapcar #'(lambda (slot-spec)
-                                 (if (consp slot-spec)
-                                     (cadr slot-spec)
-                                     slot-spec))
-                             slots))
-         (supplied-vars (mapcar #'(lambda (slot)
-                                    (gensym (symbol-name
-                                             (symbol-concat slot '#:-p))))
-                                slot-names))
-         (key-args (mapcar #'(lambda (slot-spec supplied)
-                               `(,slot-spec nil ,supplied))
-                           slots supplied-vars))
-         (key-arg-alist (mapcar #'cons slot-names supplied-vars)))
+  "Each element of SLOTS is either a symbol naming a slot
+or (SLOT-NAME ARGUMENT-ONLY-P)."
```

Comment:
the previous default was more intuitive: by default, it is a slot, you need to explicitly specify, that it is not (by putting nil as a second argument)

so argument-only-p -> slotp and proper destructuring with &optional (slotp t)

Reply:
This is only used in one heavily commented place. I think keeping the macro as simple as possible makes more sense in this case.

> so argument-only-p -> slotp and proper destructuring with &optional (slotp t)

Destructuring is not needed beyond `for (name argument-only-p) = (alexandria:ensure-list slot-spec)`. I think your comment about the destructuring part somehow got attached to an outdated version of the code.

Reply:
foo means slot (the default), but
(foo t) means *not* slot
(foo nil) means *not* *not* a slot

confirming negation (i.e T stands for "true that it is not a slot") is very weird here

Reply:
I flipped the meaning from `argument-only-p` to `slot-p`.